File: sakinah-android/app/src/main/kotlin/org/emdatra/sakinah/app/ui/Components.kt
package org.emdatra.sakinah.app.ui

import androidx.compose.animation.animateColorAsState
import androidx.compose.animation.core.animateDpAsState
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material.icons.outlined.*
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

// ===== مكوّنات نظام التصميم (مطابقة لصفحة Components في Figma) =====

enum class ButtonKind { Primary, Gold, Soft, Outline, Ghost }

@Composable fun DSButton(label: String, modifier: Modifier = Modifier, kind: ButtonKind = ButtonKind.Primary, icon: ImageVector? = null, enabled: Boolean = true, onClick: () -> Unit) {
  val c = DS.c
  val fg = when (kind) { ButtonKind.Primary -> c.textOnBrand; ButtonKind.Gold -> Color(0xFF16211F); else -> c.brandPrimary }
  val bg = when (kind) { ButtonKind.Primary -> c.brandPrimary; ButtonKind.Gold -> c.accentGold; ButtonKind.Soft -> c.brandSoft; else -> Color.Transparent }
  Row(modifier.clip(DS.shapeLg).background(bg).then(if (kind == ButtonKind.Outline) Modifier.border(1.5.dp, c.borderStrong, DS.shapeLg) else Modifier).clickable(enabled = enabled, role = Role.Button, onClick = onClick).padding(horizontal = 20.dp, vertical = 14.dp), horizontalArrangement = Arrangement.Center, verticalAlignment = Alignment.CenterVertically) {
    Text(label, style = DSType.labelMd, color = fg)
    if (icon != null) { Spacer(Modifier.width(8.dp)); Icon(icon, null, Modifier.size(16.dp), tint = fg) }
  }
}

enum class IconStyle { Outlined, Soft, GoldSoft, Glass, Brand, Plain }

@Composable fun DSIconButton(icon: ImageVector, modifier: Modifier = Modifier, style: IconStyle = IconStyle.Outlined, size: Dp = 42.dp, iconSize: Dp = 18.dp, contentDescription: String? = null, onClick: (() -> Unit)? = null) {
  val c = DS.c
  val fg = when (style) { IconStyle.Soft -> c.brandPrimary; IconStyle.GoldSoft -> c.accentGoldStrong; IconStyle.Glass -> c.textOnDark; IconStyle.Brand -> c.textOnBrand; else -> c.textPrimary }
  val bg = when (style) { IconStyle.Outlined -> c.bgSurface; IconStyle.Soft -> c.brandSoft; IconStyle.GoldSoft -> c.accentGoldSoft; IconStyle.Glass -> Color.White.copy(alpha = 0.12f); IconStyle.Brand -> c.brandPrimary; IconStyle.Plain -> Color.Transparent }
  Box(modifier.size(size).clip(CircleShape).background(bg).then(if (style == IconStyle.Outlined) Modifier.border(1.dp, c.borderSubtle, CircleShape) else Modifier).then(if (onClick != null) Modifier.clickable(role = Role.Button, onClick = onClick) else Modifier), contentAlignment = Alignment.Center) {
    Icon(icon, contentDescription, Modifier.size(iconSize), tint = fg)
  }
}

@Composable fun DSChip(label: String, on: Boolean = false, icon: ImageVector? = null, modifier: Modifier = Modifier, onClick: () -> Unit = {}) {
  val c = DS.c
  Row(modifier.clip(CircleShape).background(if (on) c.brandPrimary else c.bgSurface).then(if (!on) Modifier.border(1.dp, c.borderSubtle, CircleShape) else Modifier).clickable(onClick = onClick).padding(horizontal = 14.dp, vertical = 8.dp), verticalAlignment = Alignment.CenterVertically) {
    if (icon != null) { Icon(icon, null, Modifier.size(14.dp), tint = if (on) c.textOnBrand else c.textSecondary); Spacer(Modifier.width(6.dp)) }
    Text(label, style = DSType.labelSm, color = if (on) c.textOnBrand else c.textSecondary)
  }
}

/** بطاقة سطح بزاوية 24 وظلّ خفيف */
@Composable fun DSCard(modifier: Modifier = Modifier, padding: Dp = 20.dp, radius: Dp = DS.Radius.xl, onClick: (() -> Unit)? = null, content: @Composable ColumnScope.() -> Unit) {
  val c = DS.c; val shape = RoundedCornerShape(radius)
  Column(modifier.shadow(if (c.isDark) 0.dp else 8.dp, shape, ambientColor = c.shadow.copy(alpha = 0.10f), spotColor = c.shadow.copy(alpha = 0.12f)).clip(shape).background(c.bgSurface).then(if (onClick != null) Modifier.clickable(onClick = onClick) else Modifier).padding(padding), content = content)
}
/** بلاطة بحدّ خفيف بلا ظلّ */
@Composable fun DSTile(modifier: Modifier = Modifier, padding: Dp = 14.dp, radius: Dp = DS.Radius.lg, onClick: (() -> Unit)? = null, content: @Composable ColumnScope.() -> Unit) {
  val c = DS.c; val shape = RoundedCornerShape(radius)
  Column(modifier.clip(shape).background(c.bgSurface).border(1.dp, c.borderSubtle, shape).then(if (onClick != null) Modifier.clickable(onClick = onClick) else Modifier).padding(padding), content = content)
}
/** بطاقة «الليل» المتدرّجة مع حلقات ونجوم */
@Composable fun NightCard(modifier: Modifier = Modifier, radius: Dp = DS.Radius.xxl, padding: PaddingValues = PaddingValues(horizontal = 24.dp, vertical = 22.dp), content: @Composable ColumnScope.() -> Unit) {
  val c = DS.c; val shape = RoundedCornerShape(radius)
  Box(modifier.shadow(if (c.isDark) 0.dp else 16.dp, shape, ambientColor = c.shadow.copy(alpha = 0.18f), spotColor = c.shadow.copy(alpha = 0.22f)).clip(shape).background(DS.night)) {
    Canvas(Modifier.matchParentSize()) {
      val gold = Color(0xFFC69C3E)
      drawCircle(gold.copy(alpha = 0.35f), radius = 150.dp.toPx(), center = Offset(size.width - 30.dp.toPx(), 0f), style = Stroke(1.dp.toPx()))
      drawCircle(Color.White.copy(alpha = 0.12f), radius = 110.dp.toPx(), center = Offset(size.width - 30.dp.toPx(), 0f), style = Stroke(1.dp.toPx()))
      listOf(Triple(40f, 26f, 2f), Triple(92f, 54f, 1.5f), Triple(64f, 118f, 1f), Triple(150f, 30f, 1f)).forEach { (x, y, r) -> drawCircle(gold.copy(alpha = 0.8f), radius = r.dp.toPx(), center = Offset(size.width - x.dp.toPx(), y.dp.toPx())) }
    }
    Column(Modifier.padding(padding), content = content)
  }
}

/** شريط تقدّم يبدأ من جهة البداية (اليمين في العربية) */
@Composable fun ProgressTrack(progress: Float, modifier: Modifier = Modifier, tint: Color = DS.c.accentGold, track: Color = Color.White.copy(alpha = 0.18f), height: Dp = 6.dp) {
  val p by animateFloatAsState(progress.coerceIn(0f, 1f), label = "progress")
  Box(modifier.fillMaxWidth().height(height).clip(CircleShape).background(track)) { Box(Modifier.fillMaxWidth(p.coerceAtLeast(0.02f)).fillMaxHeight().clip(CircleShape).background(tint)) }
}
/** حلقة تقدّم دائرية */
@Composable fun RingProgress(progress: Float, modifier: Modifier = Modifier, tint: Color = DS.c.accentGold, track: Color = DS.c.borderSubtle, stroke: Dp = 6.dp) {
  val p by animateFloatAsState(progress.coerceIn(0f, 1f), label = "ring")
  Canvas(modifier) {
    val w = stroke.toPx(); val inset = w / 2
    drawArc(track, 0f, 360f, false, topLeft = Offset(inset, inset), size = androidx.compose.ui.geometry.Size(size.width - w, size.height - w), style = Stroke(w))
    drawArc(tint, -90f, 360f * p.coerceAtLeast(0.001f), false, topLeft = Offset(inset, inset), size = androidx.compose.ui.geometry.Size(size.width - w, size.height - w), style = Stroke(w, cap = StrokeCap.Round))
  }
}

@Composable fun DSSectionHead(title: String, modifier: Modifier = Modifier, link: String? = null, onLink: (() -> Unit)? = null) {
  val c = DS.c
  Row(modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
    Text(title, style = DSType.headingMd, color = c.textPrimary)
    Spacer(Modifier.weight(1f))
    if (link != null) Row(Modifier.clickable(enabled = onLink != null) { onLink?.invoke() }, verticalAlignment = Alignment.CenterVertically) { Text(link, style = DSType.labelSm, color = c.brandPrimary); Icon(Icons.Filled.ChevronLeft, null, Modifier.size(14.dp), tint = c.brandPrimary) }
  }
}

/** صفّ قائمة: أيقونة في قرص خفيف + عنوان (+ وصف) + عنصر في النهاية */
@Composable fun DSRow(icon: ImageVector, title: String, subtitle: String? = null, iconStyle: IconStyle = IconStyle.Soft, modifier: Modifier = Modifier, onClick: (() -> Unit)? = null, trailing: @Composable RowScope.() -> Unit = { Icon(Icons.Filled.ChevronLeft, null, Modifier.size(16.dp), tint = DS.c.textTertiary) }) {
  val c = DS.c
  Row(modifier.fillMaxWidth().then(if (onClick != null) Modifier.clickable(onClick = onClick) else Modifier).padding(horizontal = 12.dp, vertical = 10.dp), verticalAlignment = Alignment.CenterVertically) {
    DSIconButton(icon, style = iconStyle, size = 40.dp, iconSize = 17.dp)
    Spacer(Modifier.width(12.dp))
    Column(Modifier.weight(1f)) { Text(title, style = DSType.labelMd, color = c.textPrimary); if (subtitle != null) Text(subtitle, style = DSType.labelXs, color = c.textSecondary) }
    trailing()
  }
}

/** شريط عنوان: رجوع يمينًا، عنوان في الوسط، إجراء يسارًا */
@Composable fun DSNavBar(title: String, subtitle: String? = null, onBack: (() -> Unit)? = null, trailing: @Composable () -> Unit = {}) {
  val c = DS.c
  Row(Modifier.fillMaxWidth().padding(horizontal = 20.dp, vertical = 4.dp), verticalAlignment = Alignment.CenterVertically) {
    if (onBack != null) DSIconButton(Icons.Filled.ChevronRight, iconSize = 18.dp, contentDescription = "رجوع", onClick = onBack) else Spacer(Modifier.size(42.dp))
    Column(Modifier.weight(1f), horizontalAlignment = Alignment.CenterHorizontally) { Text(title, style = DSType.headingMd, color = c.textPrimary, maxLines = 1); if (subtitle != null) Text(subtitle, style = DSType.labelXs, color = c.textSecondary, maxLines = 1) }
    Box(Modifier.widthIn(min = 42.dp), contentAlignment = Alignment.Center) { trailing() }
  }
}

enum class AppTab(val title: String, val icon: ImageVector, val iconSelected: ImageVector) {
  Prayer("الصلاة", Icons.Outlined.Home, Icons.Filled.Home), Qibla("القبلة", Icons.Outlined.Explore, Icons.Filled.Explore), Mushaf("المصحف", Icons.Outlined.MenuBook, Icons.Filled.MenuBook), Adhkar("الأذكار", Icons.Outlined.Grain, Icons.Filled.Grain), More("المزيد", Icons.Outlined.MoreHoriz, Icons.Filled.MoreHoriz)
}
/** شريط التبويبات الخمسة */
@Composable fun DSTabBar(selected: AppTab, onSelect: (AppTab) -> Unit) {
  val c = DS.c
  Column(Modifier.fillMaxWidth().background(c.bgSurface)) {
    Box(Modifier.fillMaxWidth().height(0.5.dp).background(c.borderSubtle))
    Row(Modifier.fillMaxWidth().navigationBarsPadding().padding(horizontal = 8.dp, vertical = 6.dp)) {
      AppTab.entries.forEach { tab ->
        val on = tab == selected
        val tint by animateColorAsState(if (on) c.brandPrimary else c.textTertiary, label = "tab")
        Column(Modifier.weight(1f).clip(DS.shapeMd).clickable(interactionSource = remember { MutableInteractionSource() }, indication = null, role = Role.Tab, onClick = { onSelect(tab) }).padding(vertical = 6.dp), horizontalAlignment = Alignment.CenterHorizontally) {
          Icon(if (on) tab.iconSelected else tab.icon, tab.title, Modifier.size(24.dp), tint = tint)
          Spacer(Modifier.height(3.dp))
          Text(tab.title, style = DSType.labelXs, color = tint)
        }
      }
    }
  }
}

/** حبّة معلومات (التاريخ الهجري) */
@Composable fun DSPill(icon: ImageVector, text: String, secondary: String? = null, modifier: Modifier = Modifier) {
  val c = DS.c
  Row(modifier.clip(CircleShape).background(c.bgSubtle).padding(horizontal = 14.dp, vertical = 8.dp), verticalAlignment = Alignment.CenterVertically) {
    Icon(icon, null, Modifier.size(14.dp), tint = c.accentGoldStrong); Spacer(Modifier.width(8.dp))
    Text(text, style = DSType.labelSm, color = c.textSecondary)
    if (secondary != null) { Text("  ·  ", style = DSType.labelSm, color = c.textTertiary); Text(secondary, style = DSType.labelSm, color = c.textTertiary) }
  }
}
@Composable fun DSBadge(text: String, fg: Color = DS.c.textOnBrand, bg: Color = DS.c.brandPrimary) {
  Text(text, style = DSType.labelXs, color = fg, modifier = Modifier.clip(CircleShape).background(bg).padding(horizontal = 8.dp, vertical = 3.dp))
}
@Composable fun DSQuickTile(icon: ImageVector, title: String, modifier: Modifier = Modifier, onClick: () -> Unit) {
  val c = DS.c
  DSTile(modifier, padding = 0.dp, onClick = onClick) {
    Column(Modifier.fillMaxWidth().padding(vertical = 14.dp, horizontal = 6.dp), horizontalAlignment = Alignment.CenterHorizontally) {
      Icon(icon, null, Modifier.size(22.dp), tint = c.brandPrimary); Spacer(Modifier.height(8.dp)); Text(title, style = DSType.labelSm, color = c.textPrimary, maxLines = 1)
    }
  }
}
@Composable fun DSStatTile(value: String, label: String, modifier: Modifier = Modifier) {
  val c = DS.c
  DSTile(modifier, padding = 0.dp) {
    Column(Modifier.fillMaxWidth().padding(vertical = 12.dp, horizontal = 8.dp), horizontalAlignment = Alignment.CenterHorizontally) {
      Text(value, style = DSType.numericLg, color = c.textPrimary, maxLines = 1); Text(label, style = DSType.labelXs, color = c.textSecondary)
    }
  }
}
/** مفتاح تبديل بلون الهوية */
@Composable fun DSToggle(checked: Boolean, onChange: (Boolean) -> Unit, modifier: Modifier = Modifier) {
  val c = DS.c
  val bg by animateColorAsState(if (checked) c.brandPrimary else c.borderStrong, label = "toggle")
  val pad by animateDpAsState(if (checked) 21.dp else 3.dp, label = "knob")
  Box(modifier.size(46.dp, 28.dp).clip(CircleShape).background(bg).clickable(role = Role.Switch) { onChange(!checked) }.padding(start = pad, top = 3.dp)) {
    Box(Modifier.size(22.dp).clip(CircleShape).background(if (checked) c.textOnBrand else c.bgSurface))
  }
}
/** صفّ إعداد: عنوان ووصف + مفتاح تبديل */
@Composable fun DSToggleRow(title: String, subtitle: String? = null, checked: Boolean, onChange: (Boolean) -> Unit) {
  val c = DS.c
  Row(Modifier.fillMaxWidth().clickable { onChange(!checked) }.padding(horizontal = 14.dp, vertical = 10.dp), verticalAlignment = Alignment.CenterVertically) {
    Column(Modifier.weight(1f)) { Text(title, style = DSType.labelMd, color = c.textPrimary); if (subtitle != null) Text(subtitle, style = DSType.labelXs, color = c.textSecondary) }
    DSToggle(checked, onChange)
  }
}
@Composable fun DSDivider(modifier: Modifier = Modifier) { Box(modifier.fillMaxWidth().height(1.dp).background(DS.c.borderSubtle)) }
@Composable fun Dots(count: Int, active: Int) {
  val c = DS.c
  Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) { repeat(count) { i -> val w by animateDpAsState(if (i == active) 24.dp else 8.dp, label = "dot"); Box(Modifier.size(w, 8.dp).clip(CircleShape).background(if (i == active) c.accentGold else c.borderStrong)) } }
}

File: sakinah-native/App/Sakinah/Components.swift
import SwiftUI

// MARK: - الأزرار
struct DSButtonLabel: View {
  enum Kind { case primary, gold, soft, outline, ghost }
  let title: String
  var kind: Kind = .primary
  var icon: String? = nil
  var fill: Bool = true
  var body: some View {
    HStack(spacing: 8) {
      Text(title).font(DS.F.labelMd)
      if let icon { Image(systemName: icon).font(.system(size: 14, weight: .semibold)) }
    }
    .foregroundStyle(fg)
    .padding(.vertical, 14).padding(.horizontal, 20)
    .frame(maxWidth: fill ? .infinity : nil)
    .background(bg, in: RoundedRectangle(cornerRadius: DS.Radius.lg, style: .continuous))
    .overlay { if kind == .outline { RoundedRectangle(cornerRadius: DS.Radius.lg, style: .continuous).stroke(DS.C.borderStrong, lineWidth: 1.5) } }
    .contentShape(Rectangle())
  }
  private var fg: Color { switch kind { case .primary: return DS.C.textOnBrand; case .gold: return Color(hex: 0x16211F); default: return DS.C.brandPrimary } }
  private var bg: Color { switch kind { case .primary: return DS.C.brandPrimary; case .gold: return DS.C.accentGold; case .soft: return DS.C.brandSoft; default: return .clear } }
}
struct DSButton: View {
  let title: String
  var kind: DSButtonLabel.Kind = .primary
  var icon: String? = nil
  var fill: Bool = true
  let action: () -> Void
  var body: some View { Button(action: action) { DSButtonLabel(title: title, kind: kind, icon: icon, fill: fill) }.buttonStyle(.plain) }
}

// MARK: - زر أيقونة دائري
struct DSIcon: View {
  enum Style { case outlined, soft, gold, glass, brand, plain }
  let systemName: String
  var style: Style = .outlined
  var size: CGFloat = 42
  var iconSize: CGFloat = 18
  var body: some View {
    Image(systemName: systemName).font(.system(size: iconSize, weight: .medium)).foregroundStyle(fg)
      .frame(width: size, height: size)
      .background(bg, in: Circle())
      .overlay { if style == .outlined { Circle().stroke(DS.C.borderSubtle, lineWidth: 1) } }
      .contentShape(Circle())
  }
  private var fg: Color { switch style { case .soft: return DS.C.brandPrimary; case .gold: return DS.C.accentGoldStrong; case .glass: return DS.C.textOnDark; case .brand: return DS.C.textOnBrand; default: return DS.C.textPrimary } }
  private var bg: Color { switch style { case .outlined: return DS.C.bgSurface; case .soft: return DS.C.brandSoft; case .gold: return DS.C.accentGoldSoft; case .glass: return Color.white.opacity(0.12); case .brand: return DS.C.brandPrimary; case .plain: return .clear } }
}
struct DSIconButton: View {
  let systemName: String
  var style: DSIcon.Style = .outlined
  var size: CGFloat = 42
  var iconSize: CGFloat = 18
  var label: String? = nil
  let action: () -> Void
  var body: some View { Button(action: action) { DSIcon(systemName: systemName, style: style, size: size, iconSize: iconSize) }.buttonStyle(.plain).accessibilityLabel(label ?? systemName) }
}

// MARK: - شريحة
struct DSChip: View {
  let title: String
  var on = false
  var icon: String? = nil
  var action: (() -> Void)? = nil
  var body: some View {
    Button { action?() } label: {
      HStack(spacing: 6) { if let icon { Image(systemName: icon).font(.system(size: 12, weight: .semibold)) }; Text(title).font(DS.F.labelSm) }
        .foregroundStyle(on ? DS.C.textOnBrand : DS.C.textSecondary)
        .padding(.vertical, 8).padding(.horizontal, 14)
        .background(on ? DS.C.brandPrimary : DS.C.bgSurface, in: Capsule())
        .overlay { if !on { Capsule().stroke(DS.C.borderSubtle, lineWidth: 1) } }
    }.buttonStyle(.plain)
  }
}

// MARK: - البطاقات والأسطح
extension View {
  /// بطاقة سطح بيضاء بزاوية 24 وظلّ خفيف
  func dsCard(padding: CGFloat = 20, radius: CGFloat = DS.Radius.xl) -> some View {
    self.padding(padding)
      .background(DS.C.bgSurface, in: RoundedRectangle(cornerRadius: radius, style: .continuous))
      .shadow(color: DS.C.shadowCard, radius: 12, x: 0, y: 4)
  }
  /// بطاقة بحدّ خفيف بلا ظلّ (للبلاطات الصغيرة)
  func dsTile(padding: CGFloat = 14, radius: CGFloat = DS.Radius.lg) -> some View {
    self.padding(padding)
      .background(DS.C.bgSurface, in: RoundedRectangle(cornerRadius: radius, style: .continuous))
      .overlay { RoundedRectangle(cornerRadius: radius, style: .continuous).stroke(DS.C.borderSubtle, lineWidth: 1) }
  }
  /// بطاقة «الليل» المتدرّجة مع حلقات وزخرفة نجوم
  func nightCard(radius: CGFloat = DS.Radius.xxl) -> some View {
    self.background { ZStack { DS.nightGradient; NightDecor() } }
      .clipShape(RoundedRectangle(cornerRadius: radius, style: .continuous))
      .shadow(color: DS.C.shadowFloat, radius: 20, x: 0, y: 12)
  }
  /// إخفاء شريط التبويبات النظامي (نستخدم DSTabBar)
  func hiddenSystemTabBar() -> some View { self.toolbar(.hidden, for: .tabBar) }
}

/// حلقات ذهبية شفافة ونجوم صغيرة (تُرسم بإحداثيات مطلقة كي لا تنعكس مع الاتجاه)
struct NightDecor: View {
  var body: some View {
    GeometryReader { g in
      ZStack {
        Circle().stroke(DS.C.accentGold.opacity(0.35), lineWidth: 1).frame(width: 300, height: 300).position(x: 30, y: 0)
        Circle().stroke(Color.white.opacity(0.12), lineWidth: 1).frame(width: 220, height: 220).position(x: 30, y: 0)
        ForEach(Array([(40.0, 26.0, 4.0), (92.0, 54.0, 3.0), (64.0, 118.0, 2.0), (150.0, 30.0, 2.0)].enumerated()), id: \.offset) { _, s in
          Circle().fill(DS.C.accentGold.opacity(0.8)).frame(width: s.2, height: s.2).position(x: s.0, y: s.1)
        }
      }
      .frame(width: g.size.width, height: g.size.height)
    }
    .allowsHitTesting(false)
  }
}

// MARK: - رأس قسم
struct DSSectionHead<Link: View>: View {
  let title: String
  @ViewBuilder var link: () -> Link
  init(_ title: String, @ViewBuilder link: @escaping () -> Link) { self.title = title; self.link = link }
  var body: some View { HStack { Text(title).font(DS.F.headingMd).foregroundStyle(DS.C.textPrimary); Spacer(); link() } }
}
extension DSSectionHead where Link == EmptyView { init(_ title: String) { self.init(title) { EmptyView() } } }
/// رابط قسم «العنوان ›»
struct DSLinkLabel: View {
  let title: String
  var body: some View { HStack(spacing: 2) { Text(title).font(DS.F.labelSm); Image(systemName: "chevron.forward").font(.system(size: 11, weight: .semibold)) }.foregroundStyle(DS.C.brandPrimary) }
}

// MARK: - شريط تقدّم (يبدأ من جهة البداية: اليمين في العربية)
struct ProgressTrack: View {
  let progress: Double
  var tint: Color = DS.C.accentGold
  var track: Color = Color.white.opacity(0.18)
  var height: CGFloat = 6
  var body: some View {
    GeometryReader { g in
      HStack(spacing: 0) { Capsule().fill(tint).frame(width: max(height, g.size.width * min(1, max(0, progress)))); Spacer(minLength: 0) }
    }
    .frame(height: height)
    .background(track, in: Capsule())
    .animation(.easeInOut(duration: 0.4), value: progress)
  }
}

/// حلقة تقدّم دائرية
struct RingProgress: View {
  let progress: Double
  var tint: Color = DS.C.accentGold
  var track: Color = DS.C.borderSubtle
  var lineWidth: CGFloat = 6
  var body: some View {
    ZStack {
      Circle().stroke(track, lineWidth: lineWidth)
      Circle().trim(from: 0, to: min(1, max(0.001, progress))).stroke(tint, style: StrokeStyle(lineWidth: lineWidth, lineCap: .round)).rotationEffect(.degrees(-90))
    }
    .animation(.easeInOut(duration: 0.5), value: progress)
  }
}

// MARK: - صفّ قائمة
struct DSRow<Trailing: View>: View {
  let icon: String
  var iconStyle: DSIcon.Style = .soft
  let title: String
  var subtitle: String? = nil
  @ViewBuilder var trailing: () -> Trailing
  var body: some View {
    HStack(spacing: 12) {
      DSIcon(systemName: icon, style: iconStyle, size: 40, iconSize: 17)
      VStack(alignment: .leading, spacing: 1) {
        Text(title).font(DS.F.labelMd).foregroundStyle(DS.C.textPrimary)
        if let subtitle { Text(subtitle).font(DS.F.labelXs).foregroundStyle(DS.C.textSecondary) }
      }
      Spacer(minLength: 8)
      trailing()
    }
    .padding(.vertical, 10).padding(.horizontal, 12)
    .contentShape(Rectangle())
  }
}
extension DSRow where Trailing == DSChevron { init(icon: String, iconStyle: DSIcon.Style = .soft, title: String, subtitle: String? = nil) { self.init(icon: icon, iconStyle: iconStyle, title: title, subtitle: subtitle) { DSChevron() } } }
struct DSChevron: View { var body: some View { Image(systemName: "chevron.forward").font(.system(size: 13, weight: .semibold)).foregroundStyle(DS.C.textTertiary) } }

// MARK: - شريط عنوان مخصّص (عنوان في الوسط، رجوع يمينًا، إجراء يسارًا)
struct DSNavBar<Trailing: View>: View {
  let title: String
  var subtitle: String? = nil
  var back: (() -> Void)? = nil
  @ViewBuilder var trailing: () -> Trailing
  var body: some View {
    HStack(spacing: 8) {
      if let back { DSIconButton(systemName: "chevron.forward", iconSize: 16, label: "رجوع", action: back) } else { Color.clear.frame(width: 42, height: 42) }
      VStack(spacing: 0) {
        Text(title).font(DS.F.headingMd).foregroundStyle(DS.C.textPrimary).lineLimit(1)
        if let subtitle { Text(subtitle).font(DS.F.labelXs).foregroundStyle(DS.C.textSecondary).lineLimit(1) }
      }.frame(maxWidth: .infinity)
      trailing().frame(minWidth: 42)
    }
    .padding(.horizontal, 20).padding(.vertical, 4)
  }
}
extension DSNavBar where Trailing == EmptyView { init(title: String, subtitle: String? = nil, back: (() -> Void)? = nil) { self.init(title: title, subtitle: subtitle, back: back) { EmptyView() } } }

// MARK: - شريط التبويبات
enum AppTab: Int, CaseIterable, Identifiable {
  case prayer, qibla, mushaf, adhkar, more
  var id: Int { rawValue }
  var title: String { switch self { case .prayer: return "الصلاة"; case .qibla: return "القبلة"; case .mushaf: return "المصحف"; case .adhkar: return "الأذكار"; case .more: return "المزيد" } }
  var icon: String { switch self { case .prayer: return "house"; case .qibla: return "location.north.circle"; case .mushaf: return "book"; case .adhkar: return "circle.hexagongrid"; case .more: return "ellipsis" } }
  var iconSelected: String { switch self { case .prayer: return "house.fill"; case .qibla: return "location.north.circle.fill"; case .mushaf: return "book.fill"; case .adhkar: return "circle.hexagongrid.fill"; case .more: return "ellipsis" } }
}
struct DSTabBar: View {
  @Binding var selection: AppTab
  var body: some View {
    HStack(spacing: 0) {
      ForEach(AppTab.allCases) { tab in
        let on = selection == tab
        Button { withAnimation(.snappy(duration: 0.2)) { selection = tab } } label: {
          VStack(spacing: 4) {
            Image(systemName: on ? tab.iconSelected : tab.icon).font(.system(size: 22, weight: on ? .semibold : .regular)).frame(height: 26)
            Text(tab.title).font(DS.F.labelXs)
          }
          .foregroundStyle(on ? DS.C.brandPrimary : DS.C.textTertiary)
          .frame(maxWidth: .infinity).padding(.top, 8).padding(.bottom, 4)
          .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .accessibilityLabel(tab.title).accessibilityAddTraits(on ? .isSelected : [])
      }
    }
    .padding(.horizontal, 8)
    .background(DS.C.bgSurface.ignoresSafeArea(edges: .bottom))
    .overlay(alignment: .top) { Rectangle().fill(DS.C.borderSubtle).frame(height: 0.5) }
  }
}

// MARK: - عناصر صغيرة
/// حبّة التاريخ الهجري
struct DSPill: View {
  let icon: String
  let text: String
  var secondary: String? = nil
  var body: some View {
    HStack(spacing: 8) {
      Image(systemName: icon).font(.system(size: 13, weight: .medium)).foregroundStyle(DS.C.accentGoldStrong)
      Text(text).font(DS.F.labelSm).foregroundStyle(DS.C.textSecondary)
      if let secondary { Text("·").foregroundStyle(DS.C.textTertiary); Text(secondary).font(DS.F.labelSm).foregroundStyle(DS.C.textTertiary) }
    }
    .padding(.vertical, 8).padding(.horizontal, 14)
    .background(DS.C.bgSubtle, in: Capsule())
  }
}
/// شارة نصّية صغيرة
struct DSBadge: View {
  let text: String
  var fg: Color = DS.C.textOnBrand
  var bg: Color = DS.C.brandPrimary
  var body: some View { Text(text).font(DS.F.labelXs).foregroundStyle(fg).padding(.vertical, 3).padding(.horizontal, 8).background(bg, in: Capsule()) }
}
/// بلاطة إجراء سريع (أيقونة فوق تسمية)
struct DSQuickTile: View {
  let icon: String
  let title: String
  var body: some View {
    VStack(spacing: 8) {
      Image(systemName: icon).font(.system(size: 20, weight: .medium)).foregroundStyle(DS.C.brandPrimary)
      Text(title).font(DS.F.labelSm).foregroundStyle(DS.C.textPrimary).lineLimit(1).minimumScaleFactor(0.8)
    }
    .frame(maxWidth: .infinity).padding(.vertical, 14).padding(.horizontal, 6)
    .dsTile(padding: 0)
  }
}
/// بلاطة إحصاء (رقم فوق تسمية)
struct DSStatTile: View {
  let value: String
  let label: String
  var body: some View {
    VStack(spacing: 2) {
      Text(value).font(DS.F.numericLg).foregroundStyle(DS.C.textPrimary).monospacedDigit().lineLimit(1).minimumScaleFactor(0.7)
      Text(label).font(DS.F.labelXs).foregroundStyle(DS.C.textSecondary)
    }
    .frame(maxWidth: .infinity).padding(.vertical, 12).padding(.horizontal, 8)
    .dsTile(padding: 0)
  }
}
/// نمط مفتاح التبديل بلون الهوية
struct DSToggleStyle: ToggleStyle {
  func makeBody(configuration: Configuration) -> some View {
    HStack {
      configuration.label
      Spacer()
      ZStack(alignment: configuration.isOn ? .trailing : .leading) {
        Capsule().fill(configuration.isOn ? DS.C.brandPrimary : DS.C.borderStrong).frame(width: 46, height: 28)
        Circle().fill(configuration.isOn ? DS.C.textOnBrand : DS.C.bgSurface).frame(width: 22, height: 22).padding(3)
      }
      .animation(.snappy(duration: 0.2), value: configuration.isOn)
      .onTapGesture { configuration.isOn.toggle() }
    }
  }
}
